test/_init_tests.js

```diff
@@ -207,10 +207,31 @@ describe.configure = function() {
   return new TestConfig(describe);
 };
 
+const realIt = it;
+it = function(message, runnable) { // eslint-disable-line no-native-reassign, no-undef
+  if (!runnable || !runnable.constructor
+      || runnable.constructor.name !== 'GeneratorFunction') {
+    return realIt(message, runnable);
+  }
+  return realIt(message, done => {
+    const iterator = runnable();
+    const interval = setInterval(() => {
```

Comment:
This is awesome! Would fix tests with callback hell and can simplify many promise related tests! 
One question I have: I know in our code we use `setTimeout` to delay callback. are the callback in `setTimeout` micro task as well? 
Does the `setInterval` work in the following scenario:
```
param = false
promise = new Promise(resolve => {resolver = resolve});
promise.then(() => {param = true})
doSomething = () => {setTimeout(resolver)}
doSomething();
yield
expect(param).to.be.true
```